src/sprite_store.cpp

```diff
@@ -97,70 +96,68 @@ static bool ReadUtf8Text(RcdFileReader *rcd_file, size_t expected_length, char *
 			}
 		}
 	}
-	if (real_size != 0 || expected_null != 0) return false;
+	if (real_size != 0 || expected_null != 0) rcd_file->Error("UTF8: Wrong string length");
 
 	*used_size += expected_length;
-	return true;
 }
 
 /**
  * Load a TEXT data block into the object.
  * @param rcd_file Input file.
- * @return Load was successful.
  */
-bool TextData::Load(RcdFileReader *rcd_file)
+void TextData::Load(RcdFileReader *rcd_file)
 {
+	rcd_file->CheckVersion(3);
 	char buffer[64*1024]; // Arbitrary sized block of temporary memory to store the text data.
 	size_t used_size = 0;
 	uint32 length = rcd_file->size;
-	if (rcd_file->version != 3) return false;
 
 	TextString strings[MAX_NUM_TEXT_STRINGS];
 	uint used_strings = 0;
 	while (length > 0) {
-		if (used_strings >= lengthof(strings)) return false; // Too many text strings.
+		if (used_strings >= lengthof(strings)) rcd_file->Error("Too many text strings");
 
-		if (length < 3) return false;
+		if (length < 3) rcd_file->Error("Length too short for string header");
 		uint16 str_length   = rcd_file->GetUInt16();
 		uint8  ident_length = rcd_file->GetUInt8();
 
-		if (static_cast<uint32>(str_length) > length) return false; // String does not fit in the block.
+		if (static_cast<uint32>(str_length) > length) rcd_file->Error("String does not fit into block");
```

Comment:
```suggestion
		if (static_cast<uint32>(str_length) > length) rcd_file->Error("String does not fit in the block");
```
"into" expresses movement, whereas the verb "fit" is not about moving things, but just measuring or checking.
To use "into", you could for example use "String cannot be inserted into the block", as "insert" is moving something.